tools/automator/Dockerfile

```diff
@@ -0,0 +1,5 @@
+FROM gcr.io/k8s-prow/pr-creator:latest-root as deps
+FROM gcr.io/istio-testing/build-tools:master-latest
```

Comment:
While this works, I am not fond of using `build-tools:master-latest` for branches. Maybe we can use cloud build to generate images for all branches or use `build-tools` and pull in the dependencies in the image.

Reply:
@howardjohn - when you have a moment can I get your thoughts.

Reply:
Would it make sense to just add these to the build-tools containers? Then we completely avoid this issue.

Also I think we really only need this on master. We shouldn't be changing the API after a release ideally. At least not for 1.4, in the future maybe it would be useful for 1.5

Reply:
> Would it make sense to just add these to the build-tools containers? Then we completely avoid this issue.

I am cool with that. The `pr-creator` is a useful, all-round tool so would be beneficial to add to the `build-tools` image. For this script itself, I can clone it in through `extra_refs`.

> Also I think we really only need this on master. We shouldn't be changing the API after a release ideally. At least not for 1.4, in the future maybe it would be useful for 1.5

Okay, I will dial it back to only `master` until/if it makes sense to extend to `release` branches. Also, if we are going to use the `build-tools` image directly we can leverage your job generator as well which is a plus for maintenance.

Reply:
For what it's worth the job generated supports a custom image per job as well so I think you could use it anyways, although we may need to add support for adding the GitHub mounts

Reply:
Nice, that is good to know. On second thought, I did see that the `image` field could be specified in the config. 

As an aside, at some point we should consider coalescing the logic of the 2 job generators, possibly exposed as different subcommands. Your is an actual generator (i.e. `generate`), while mine is a translator (i.e. `translate`). At the point where I can get rolling and make progress on https://github.com/istio/test-infra/issues/2161 (since we probably want a unified schema) might be a good point to evaluate this.